src/workerd/jsg/web-idl.h

```diff
@@ -15,13 +15,43 @@
 
 namespace workerd::jsg::webidl {
 
+// =======================================================================================
+// Base detection concepts and helpers
+
+// True if T has a JSG_KIND static member (i.e., is a JSG type).
+template <typename T>
+concept HasJsgKind = requires { T::JSG_KIND; };
+
+// Helper to detect and unwrap Ref<T> types
+template <typename T>
+struct RefTraits_ {
+  static constexpr bool isRef = false;
+};
+template <typename T>
+struct RefTraits_<Ref<T>> {
+  static constexpr bool isRef = true;
+  using Type = T;
+};
+
+template <typename T>
+concept IsRef = RefTraits_<T>::isRef;
+
+template <IsRef T>
+using RefType = typename RefTraits_<T>::Type;
```

Comment:
nit: RefType is currently unused – I'm fine with keeping it for future use, but let's add a comment if that's intentional.

Reply:
Actually, we do use `RefTraits_<T>::Type` in isRefToResource_() below, so while it is unused we'd probably want to use it there.